trollmoves/s3downloader.py

```diff
@@ -0,0 +1,371 @@
+        return True
+
+
+class FilePublisher(Thread):
+
+    """A publisher for result files listening to a publish queue.
+    Publishes the files via posttroll"""
+
+    def __init__(self, queue, nameservers):
+        Thread.__init__(self)
+        self.loop = True
+        self.queue = queue
+        self.service_name = 's3downloader'
+        self.nameservers = nameservers
+
+    def stop(self):
+        """Stops the file publisher."""
+        self.loop = False
+        self.queue.put("STOP")
+
+    def _publish_message(self, retv, publisher):
+        if not self.loop:
+            return False
+        if retv is not None:
+            LOGGER.info("Publish as service: %s", self.service_name)
+            LOGGER.info("Publish the files...")
+            publisher.send(retv)
+        return True
+
+    def run(self):
+        """Run the publish loop getting messages from the publish queue."""
+        try:
+            LOGGER.debug("Using service_name: {} with nameservers {}".format(self.service_name, self.nameservers))
+            with Publish(self.service_name, 0, nameservers=self.nameservers) as publisher:
+                for retv in self.queue.get():
+                    if not self._publish_message(retv, publisher):
+                        break
+
+        except KeyboardInterrupt:
+            LOGGER.info("Received keyboard interrupt. Shutting down")
+            raise
+
+
+class s3downloader():
+
+    def __init__(self, cmd_args):
+        self.config = None
+        self.cmd_args = cmd_args
+        self.listener_queue = queue.Queue()
+        self.publisher_queue = queue.Queue()
+        self.listener = None
+        self.publisher = None
+
+    def _stop(self):
+        self.listener.stop()
+        self.publisher.stop()
+        LOGGER.info("Exiting s3downloader.")
+
+    def start(self):
+        """Start the threads for listening and publishing messages and handeling the download."""
+        LOGGER.info("Starting up.")
+        self.listener = Listener(self.listener_queue, self.config,
+                                 subscribe_nameserver=self.cmd_args.subscribe_nameserver)
+        self.listener.start()
+
+        self.publisher = FilePublisher(self.publisher_queue, self.cmd_args.nameservers)
+        self.publisher.start()
+
+        self._read_from_queue()
+
+        self._stop()
+
+    def read_config(self, debug=True):
+        """Read the config file from cmd args.
+        """
+        if self.cmd_args.config_file and os.path.exists(self.cmd_args.config_file):
+            with open(self.cmd_args.config_file, 'r') as stream:
+                try:
+                    self.config = yaml.safe_load(stream)
+                    if debug:
+                        import pprint
+                        pp = pprint.PrettyPrinter(indent=4)
+                        pp.pprint(self.config)
+                except FileNotFoundError:
+                    print("Could not find you config file:", self.cmd_args.config_file)
```

Comment:
```suggestion
                    LOGGER.error("Could not find you config file: %s", self.cmd_args.config_file)
```

Reply:
This is before the logger is configured. Thats why I'm using print